consensus/spos/bls/subroundSignature.go

```diff
@@ -378,18 +399,24 @@ func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
 			return false
 		}
 
-		if !isMultiKeyLeader {
+		isCurrentManagedKeyLeader := idx == spos.IndexOfLeaderInConsensusGroup
+		// TODO[cleanup cns finality]: update the check
+		// with the equivalent messages feature on, signatures from all managed keys must be broadcast, as the aggregation is done by any participant
+		shouldBroadcastSignatureShare := (!isCurrentNodeMultiKeyLeader && !isFlagActive) ||
```

Comment:
what is the difference between isCurrentNodeMultiKeyLeader and isCurrentManagedKeyLeader ?

to me it seems that they should represent the same thing

and since there is an && with !isFlagActive and isFlagActive, that could be simplified/removed if isCurrentNodeMultiKeyLeader and isCurrentManagedKeyLeader indeed represent the same thing, and only go forward if current index is not leader.

Reply:
`isCurrentNodeMultiKeyLeader` = the leader is managed by this node
`isCurrentManagedKeyLeader` = that specific index is the leader, which is managed by the node

this is already removed and simplified a lot in next PRs, matching your suggestion :pray: